src/count_expressions.jl

```diff
@@ -15,6 +15,31 @@ end
 """
     count_expressions(iter::ProgramIterator)    
 
-Counts and returns the number of possible expressions in the expression iterator. The Iterator is not modified.
+Counts and returns the number of possible expressions in the expression iterator.
+!!! warning: modifies and exhausts the iterator
 """
-count_expressions(iter::ProgramIterator) = count_expressions(iter.grammar, iter.max_depth, iter.max_size, iter.sym)
+function count_expressions(iter::ProgramIterator)
```

Comment:
I think that `length(iter)` does exactly what your function is doing :)

Reply:
Cool! I didn't know that. I tested it, but it did not work out of the box:
`ERROR: MethodError: no method matching length(::BFSIterator)`
Do we need ProgramIterator to inherit from an abstract built-in julia iterator for that? It seems like julia does not recognize that `length` is referring to the number of elements of the iterator.

Reply:
Ah, that is a bit sad. I guess it's fine like this for now.

Reply:
Yes, you have to implement two additional methods: 
```julia
julia> Base.eltype(::Type{Squares}) = Int # Note that this is defined for the type

julia> Base.length(S::Squares) = S.count
```
(https://docs.julialang.org/en/v1/manual/interfaces/). Length can be, if I'm not mistaken, unknown but you have to return a specific value for that

Reply:
Hmm, so we still have to calculate the length ourselves? As there is no default implementation of `length(iter)`.

Reply:
We can rename `count_expressions` to `Base.length` to stick with the julia naming convention.

We we are removing `count_expressions` for iterators, we might as will remove `count_expressions` for (grammar, max_depth, max_size, Symbol). These cases could be then refactored like so:

```julia
#right now
@test count_expressions(g1, 1, typemax(Int), :Real) == 2

#after refactor
@test length(BFSIterator(g1, :Real, max_depth=1)) == 2
```

Reply:
Yep, let's always follow convention where possible.

Reply:
I committed the refactor from `count_expressions` to `length`.